Mathlib/Algebra/GroupPower/Lemmas.lean

```diff
@@ -645,13 +646,13 @@ theorem one_add_mul_le_pow' (Hsq : 0 ≤ a * a) (Hsq' : 0 ≤ (1 + a) * (1 + a))
       add_nonneg (mul_nonneg n.cast_nonneg (mul_nonneg Hsq H)) Hsq
     calc
       1 + (↑(n + 2) : R) * a ≤ 1 + ↑(n + 2) * a + (n * (a * a * (2 + a)) + a * a) :=
-        (le_add_iff_nonneg_right _).2 this
+        le_add_of_nonneg_right this
       _ = (1 + a) * (1 + a) * (1 + n * a) := by {
           simp only [Nat.cast_add, add_mul, mul_add, one_mul, mul_one, ← one_add_one_eq_two,
             Nat.cast_one, add_assoc, add_right_inj]
           simp only [← add_assoc, add_comm _ (↑n * a)]
-          simp only [add_assoc, add_right_inj, (n.cast_commute (_ : R)).left_comm]
-          ac_rfl }
```

Comment:
did the ac_rfl stop working? it seems a more principled and maintainable proof than simp only to me

Reply:
Yes, it didn't work. I don't know why.

Reply:
@alexjbest Do you want to have a look, or should I merge it?

Reply:
I'd like to understand why but it shouldn't hold up this PR! so please merge and I can investigate later

Reply:
minimised and reported as lean4#2413